### ISSUE TITLE
Incorrect exit code for analysis mode

### ISSUE THREAD

**Issue by CrendKing** [LinkedInAttic/naarad#341]
When --exit_code argument is specified, naarad exits with a non-zero code if error happens. naarad defines SLA failure exit code to be 1. In diff mode this works as expected. In analysis mode, the exit code is always 6 (general ERROR).

**Comment by RiteshMaheshwari** [comment]
Should be fixed in #344

**Comment by CrendKing** [comment]
Looks like SLA is reporting exit code 1 now. However, if something else is error, the exit code might still be wrong. For example, if the input file does not exist:
```
2015-09-23 18:35:52,363 - naarad - INFO - Reports will be in <output>
2015-09-23 18:35:52,363 - naarad - INFO - Available graphing modules: svg,matplotlib
2015-09-23 18:35:52,364 - naarad.utils - ERROR - Fetch/Collect failed for metric: JMETER
2015-09-23 18:35:52,389 - naarad - INFO - Reports will be in <output>
% echo $?
0
```
I'm yet to find a case to generate exit code 6.
